airflow/providers/apache/flink/operators/flink_kubernetes.py

```diff
@@ -0,0 +1,143 @@
+        If not specified, default value is ``~/.kube/config``
+    """
+
+    template_fields: Sequence[str] = ("application_file", "namespace")
+    template_ext: Sequence[str] = (".yaml", ".yml", ".json")
+    ui_color = "#f4a460"
+
+    def __init__(
+        self,
+        *,
+        application_file: str,
+        namespace: str | None = None,
+        kubernetes_conn_id: str = "kubernetes_default",
+        api_group: str = "flink.apache.org",
+        api_version: str = "v1beta1",
+        in_cluster: bool | None = None,
+        cluster_context: str | None = None,
+        config_file: str | None = None,
+        plural: str = "flinkdeployments",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.application_file = application_file
+        self.namespace = namespace
+        self.kubernetes_conn_id = kubernetes_conn_id
+        self.api_group = api_group
+        self.api_version = api_version
+        self.plural = plural
+        self.in_cluster = in_cluster
+        self.cluster_context = cluster_context
+        self.config_file = config_file
+
+    @cached_property
+    def hook(self) -> KubernetesHook:
+        hook = KubernetesHook(
+            conn_id=self.kubernetes_conn_id,
+            in_cluster=self.in_cluster,
+            config_file=self.config_file,
+            cluster_context=self.cluster_context,
+        )
+        self._patch_deprecated_k8s_settings(hook)
+        return hook
+
+    def _patch_deprecated_k8s_settings(self, hook: KubernetesHook):
+        """
+        Here we read config from core Airflow config [kubernetes] section.
+        In a future release we will stop looking at this section and require users
+        to use Airflow connections to configure KPO.
+
+        When we find values there that we need to apply on the hook, we patch special
+        hook attributes here.
+        """
+        # default for enable_tcp_keepalive is True; patch if False
+        if conf.getboolean("kubernetes", "enable_tcp_keepalive") is False:
+            hook._deprecated_core_disable_tcp_keepalive = True  # noqa
+
+        # default verify_ssl is True; patch if False.
+        if conf.getboolean("kubernetes", "verify_ssl") is False:
+            hook._deprecated_core_disable_verify_ssl = True  # noqa
+
+        # default for in_cluster is True; patch if False and no KPO param.
+        conf_in_cluster = conf.getboolean("kubernetes", "in_cluster")
+        if self.in_cluster is None and conf_in_cluster is False:
+            hook._deprecated_core_in_cluster = conf_in_cluster  # noqa
+
+        # there's no default for cluster context; if we get something (and no KPO param) patch it.
+        conf_cluster_context = conf.get("kubernetes", "cluster_context", fallback=None)
+        if not self.cluster_context and conf_cluster_context:
+            hook._deprecated_core_cluster_context = conf_cluster_context  # noqa
+
```

Comment:
@chethanuk-plutoflume Are these actually needed ? Not see them being used/referenced anywhere?

Reply:
https://github.com/apache/airflow/pull/28512/files#diff-5de054c1bea5c3b53b30769de1d806cd1d6075cc13732d984fbea70ab1501d6bR96
We are patching the hook

Reply:
Yeah I saw the function is being called, but I can't find the patched attributes being used anywhere (i.e. `hook._deprecated_core_cluster_context`)

Reply:
I think I was using _deprecated_core_cluster_context and old API in one of the internal tests, while I didn't open source. We can safely remove 119 and 120 in this case

Reply:
Got it. I think all these internal patching are not needed. I've removed them all